kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver_with_constraints.h

```diff
@@ -0,0 +1,1038 @@
+ * this information.
+ * Calculation of the reactions involves a cost very similiar to the calculation of the total residual
+ * @author Aditya Ghantasala
+ */
+template <class TSparseSpace,
+          class TDenseSpace,  //= DenseSpace<double>,
+          class TLinearSolver //= LinearSolver<TSparseSpace,TDenseSpace>
+          >
+class ResidualBasedBlockBuilderAndSolverWithConstraints
+    : public ResidualBasedBlockBuilderAndSolver<TSparseSpace, TDenseSpace, TLinearSolver>
+{
+  public:
+    ///@name Type Definitions
+    ///@{
+    KRATOS_CLASS_POINTER_DEFINITION(ResidualBasedBlockBuilderAndSolverWithConstraints);
+
+    typedef ResidualBasedBlockBuilderAndSolver<TSparseSpace, TDenseSpace, TLinearSolver> BaseType;
+
+    typedef typename BaseType::TSchemeType TSchemeType;
+
+    typedef typename BaseType::TDataType TDataType;
+
+    typedef typename BaseType::DofsArrayType DofsArrayType;
+
+    typedef typename BaseType::TSystemMatrixType TSystemMatrixType;
+
+    typedef typename BaseType::TSystemVectorType TSystemVectorType;
+
+    typedef typename BaseType::LocalSystemVectorType LocalSystemVectorType;
+
+    typedef typename BaseType::LocalSystemMatrixType LocalSystemMatrixType;
+
+    typedef typename BaseType::TSystemMatrixPointerType TSystemMatrixPointerType;
+    typedef typename BaseType::TSystemVectorPointerType TSystemVectorPointerType;
+
+    typedef Node<3> NodeType;
+
+    typedef typename BaseType::NodesArrayType NodesArrayType;
+    typedef typename BaseType::ElementsArrayType ElementsArrayType;
+    typedef typename BaseType::ConditionsArrayType ConditionsArrayType;
+
+    typedef typename BaseType::ElementsContainerType ElementsContainerType;
+
+    typedef MasterSlaveConstraint MasterSlaveConstraintType;
+    typedef typename MasterSlaveConstraint::Pointer MasterSlaveConstraintPointerType;
+
+    typedef AuxilaryGlobalMasterSlaveRelation AuxilaryGlobalMasterSlaveRelationType;
+
+    typedef PointerVectorSet<AuxilaryGlobalMasterSlaveRelationType, IndexedObject> GlobalMasterSlaveRelationContainerType;
+    typedef std::size_t SizeType;
+    typedef std::vector<SizeType> EquationIdVectorType;
+    typedef std::vector<SizeType> VectorIndexType;
+    typedef std::vector<Dof<double>::Pointer> DofsVectorType;
+    typedef Matrix MatrixType;
+    typedef Vector VectorType;
+    typedef Geometry<NodeType> GeometryType;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /** Constructor.
+     */
+    ResidualBasedBlockBuilderAndSolverWithConstraints(
+        typename TLinearSolver::Pointer pNewLinearSystemSolver)
+        : ResidualBasedBlockBuilderAndSolver<TSparseSpace, TDenseSpace, TLinearSolver>(pNewLinearSystemSolver)
+    {
+    }
+
+    /** Destructor.
+     */
+    ~ResidualBasedBlockBuilderAndSolverWithConstraints() override
+    {
+    }
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    void SetUpSystem(
```

Comment:
this function can be simplified to: 
~~~
BaseType::SetUpSystem(rModelPart);
 if(rModelPart.NumberOfMasterSlaveConstraints() > 0)
        FormulateGlobalMasterSlaveRelations(rModelPart);
~~~